src/node.rs

```diff
@@ -35,12 +41,21 @@ pub struct Node {
     /// Specify the network of this node.
     #[structopt(default_value = "2", short = "n", long = "network")]
     pub network: u16,
+    /// The listener IP of the node.
+    #[structopt(default_value = "0:0:0:0", long = "ip")]
+    pub ip: String,
```

Comment:
Why is this necessary?

Reply:
Two reasons: 
- users utilizing multiple networks might not want the node to be reachable on all of them
- we don't want test nodes to connect to the testnet, and this allows them to be `127.0.0.1`

Reply:
Oh, I see this is a `String` (ignore my earlier `Copy` comments), why not use `IpAddr`?